services/admin/src/server.ts

```diff
@@ -175,7 +175,9 @@ export function buildApp({ workspace }: { workspace: Workspace }): Application {
             errors: [
               {
                 type: result.err().kind,
-                message: JSON.stringify(result.err()),
+                message:
+                  result.err().userFriendlyMessage ||
```

Comment:
Maybe we leave user-friendly copy to the frontend, since that seems like more of a UI concern?

Reply:
Maybe the two different error cases handled here actually need two different error types?

Reply:
yeah, good point, nice to have fresh eyes on this - i iterated a couple times on direction and hadn't taken a step back to simplify

Reply:
k, just spent some time refactoring to allow the client to interpret the error codes and it ended up ballooning out of scope (and getting into the realm of "changes that might be rendered obsolete by upcoming architecture changes") - will hold off for now until we have clarity on direction for our client-server communication story.

added a separate error code for `InvalidCvrFileMode` though, to disambiguate between the two error cases

Reply:
Ok, sounds good